spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala

```diff
@@ -2201,7 +2201,8 @@ object QueryPlanSerde extends Logging with ShimQueryPlanSerde with CometExprShim
   }
 
   def nullIfWhenPrimitive(expression: Expression): Expression = if (isPrimitive(expression)) {
-    new NullIf(expression, Literal.default(expression.dataType)).child
+    val zero = Literal.default(expression.dataType)
+    If(EqualTo(expression, zero), Literal.create(null, expression.dataType), expression)
```

Comment:
?

Reply:
`NullIf` implementation was changed to use `With` that Comet does not support https://github.com/apache/spark/pull/43623